modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java

```diff
@@ -361,6 +361,66 @@ private void unregisterMBean() {
         }
     }
 
+    /**
+     *
+     */
+    public void closeAllSessions() {
+        Collection<? extends GridNioSession> sessions = srv.sessions();
+
+        for (GridNioSession ses : sessions) {
+            ClientListenerConnectionContext connCtx = ses.meta(CONN_CTX_META_KEY);
+
+            if (connCtx == null || ses.closeTime() != 0)
+                continue; // Skip non-initialized or closed session.
+
+            srv.close(ses);
+
+            if (log.isInfoEnabled()) {
+                log.info("Client session has been dropped: "
+                    + clientConnectionDescription(ses, connCtx));
+            }
+        }
+    }
+
+    /**
+     * Compose connection description string.
+     * @param ses client NIO session.
+     * @param ctx client connection context.
+     * @return connection description
+     */
+    @SuppressWarnings("StringConcatenationInsideStringBufferAppend")
+    private static String clientConnectionDescription(
+        GridNioSession ses,
+        ClientListenerConnectionContext ctx)
+    {
+        AuthorizationContext authCtx = ctx.authorizationContext();
+
+        StringBuilder sb = new StringBuilder();
+
+        if(ctx instanceof JdbcConnectionContext)
+            sb.append("JdbcClient [");
+        else if (ctx instanceof OdbcConnectionContext)
+            sb.append("OdbcClient [");
+        else
+            sb.append("ThinClient [");
+
+        InetSocketAddress rmtAddr = ses.remoteAddress();
+        InetSocketAddress locAddr = ses.localAddress();
+
+        assert rmtAddr != null;
+        assert locAddr != null;
+
+        String rmtAddrStr = rmtAddr.getHostString() + ":" + rmtAddr.getPort();
+        String locAddrStr = locAddr.getHostString() + ":" + locAddr.getPort();
+
+        sb.append("id=" + ctx.connectionId());
+        sb.append(", user=").append(authCtx == null ? "<anonymous>" : authCtx.userName());
```

Comment:
there is securityContext. See `ClientListenerAbstractConnectionContext#initClientDescriptor`